### PR TITLE
Refactors access system - access checking, desc, etc

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Refactors access checking system
every access was checked through `ACCESS in card_access` and it's bad for the time complexity.
I changed it into a kind of `if(card_access[ACCESS])` to quick check meanwhile I had to make every numeral access code a text when it's stored in a card access.
Everything is the same except when you need to store allowed access in your card. You still declare numbers in required_access in each object.

Although it's not much deal in our codebase as we really don't have much accesses in a card access, but it slightly annoyed me when I see the nature of how this works lol
also, cleaned up some bad coding

## Why It's Good For The Game
code & performance improvement

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

card console is fine

image

image

wallet gets combined acceess well

image

I can ride on a syndi mech with a wallet

image

access desc display is sane

</details>

## Changelog
:cl:
code: cleaned up some bad coding
refactor: refactored access checking system. checking access is slightly faster
/:cl:

### CODE REVIEW

File: code/modules/jobs/access.dm [hunk 1]
```diff
@@ -76,20 +77,62 @@
...
+		for(var/each_code in req_one_access)
+			if(check_access_textified(accesses_to_check, each_code)) //has an access from the single access list
 				return TRUE
 		return FALSE
 	return TRUE
 
+
+/// Note: don't merge 'access_list' in the parameter. using grant_accesses_to_card() multiple times is recommended.
+/// 'referenced_target_access' will be a reference list if you send a list of an ID card.
+/proc/grant_accesses_to_card(list/referenced_target_access, list/access_list)
```
Comment (itsmeow) [inline code/modules/jobs/access.dm]: don't explicitly type access_list if the type varies

File: code/modules/jobs/access.dm [hunk 2]
```diff
@@ -76,20 +77,62 @@
...
+	if(!access_list)
+		return
+	if(!islist(access_list)) // do not change the check to 'isnum()' because text/number comes
```
Comment (itsmeow) [inline code/modules/jobs/access.dm]: istext || isnum works

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by EvilDragonfiend** [comment]
* review addressed
* HoP fax is merged, and changed the access check code

**Review by itsmeow — APPROVED** [review]
This is real sus (I feel like it could break something) but given it's mostly a reformatting and works in testing.... ok your speedmerge privileges have been granted

**Comment by EvilDragonfiend** [comment]
thank you ❤️❤️❤️❤️❤️

**Comment by PowerfulBacon** [comment]
After doing some checks with ?compile and running both in and access check proc for 5 seconds:

|Input Size|in count|access check count|
|-|-|-|
|1|9402889|3758713|
|2|9468313|3820412|
|5|9426707|3854456|
|10|6566330|2646274|
|100|6204513|2781983|
|1000|5668724|2743046|

(I ran multiple bot commands at once for 10-1000, so there are less overall. The ratio is what is important hr)

So as for performance, this performs worse. Care should be taken when using big O notation, specifically the fact that Big O notation is a measure for how well an algorithm scales, not how well it performs. Also keep in mind we are byond and byond does weird things. In the majority of cases things that should be slower are faster due to having direct implementations in the underlying engine language (C?), which gets performance we couldn't dream of in byondland.
In addition to this, the added complexity of code and reduction of portability probably isn't worth slightly better performance on cold paths.

Part of the reason for the lower performance, is due to calling a proc. Proc calls have quite a bit of overhead when compared to functions that are almost nothing such as this.

<details>
```
/world
  loop_checks = 0

/proc/test1(i)
  var/start = world.timeofday
  var/list/a = list()
  for (var/j in 1 to i)
    a += j
  var/out
  var/count = 0
  while (world.timeofday < start + 50)
    out += (rand(1, 10) in a)
    count ++
  world.log << "Completed: [count ], Total: [out]"

/proc/test2(i)
  var/start = world.timeofday
  var/list/a = list()
  for (var/j in 1 to i)
    a["[j]"] = 1
  var/out
  var/count = 0
  while (world.timeofday < start + 50)
    out += check_access(a, "[rand(1,10)]")
    count ++
  world.log << "Completed: [count ], Total: [out]"

/proc/check_access(card, acc)
  return card["[acc]"]

/proc/main()
  test1(1)
  test2(1)
```
</details>

Considering this makes the code far more complex and will cause conflicts, I am going to recommend a revert for this PR (I'll make it when I get some free time).

**Pull request merged**